_posts/2023-02-06-tesla-ypp-2022-blogpost.md

```diff
@@ -0,0 +1,245 @@
+We are starting with more information about the global architecture.
+Later, there will be zoomed in on the architecture used for the two different applications that are built: Rock-Paper-Scissors and Google Integration.
+
+<img src="{{ '/img/2023-02-06-tesla-ypp-2022-blogpost/tesla-architecture.png' | prepend: site.baseurl }}" alt="It shows every resource used to create the whole tesla application" class="image fit" style="margin:0px auto; max-width:70%; height:70%">
+
+Above, you can see a complete diagram of the architecture we've used to accomplish our results.
+Everything within the black-lined square represents services running on AWS.
+The purple square represents our applications running on Kubernetes in the AWS EKS-Cluster.
+
+We've used GitHub and GitHub actions combined with Terraform, Helm, and Docker for deploying and CI/CD.
+
+In the context of testing, we've used Postman to test REST and Gatling for performance testing.
+
+## ROCK PAPER SCISSORS ARCHITECTURE
+
+<img src="{{ '/img/2023-02-06-tesla-ypp-2022-blogpost/rock-paper-scissors-architecture.png' | prepend: site.baseurl }}" alt="It shows every resource used to create the whole rock paper scissors application" class="image fit" style="margin:0px auto; max-width:70%; height:70%">
+
+A user will first need to authenticate themselves using Cognito before sending requests.
+
+An API Gateway was not necessary but was added for educational purposes.
+As a result, we needed to link it to our RPS(Rock-Paper-Scissors)-application via a VPC link to our Network Load Balancer (NLB) next to an Application Load Balancer (ALB) that will be connected to our RPS application.
+
+The RPS application then communicates with our Proxy-API that will (based on the active profile) communicate with one of the Tesla-APIs.
+
+Finally, the key rotation mentioned in the previous section is also included in the diagram.
+
+## GOOGLE HOME ARCHITECTURE
+
+<img src="{{ '/img/2023-02-06-tesla-ypp-2022-blogpost/google-architecture.png' | prepend: site.baseurl }}" alt="It shows every resource used to create the whole google application" class="image fit" style="margin:0px auto; max-width:70%; height:70%">
+
+The applications are deployed on Kubernetes and made accessible through Ingress and Route53.
+The Google box is linked to a Google Home account, which, in turn, is connected to our applications.
+The Proxy API is communicating with the Tesla APIs
+
+## Used Technologies
+
+The development of our project required the use of a variety of technologies to ensure a successful outcome.
+Here's a brief overview of each technology and why it was chosen.
+
+1. AWS: AWS was chosen as the cloud platform for our project because of its scalability, efficiency, and security benefits.
+By leveraging the resources offered by AWS, we were able to take advantage of the many benefits of cloud computing.
+
+2. Terraform: Terraform is an open-source tool used for infrastructure as code.
```

Comment:
"open-source" is currently under debate... Current BSL license is not a standard open-source license.